src/pyspiffe/bundle/x509_bundle/x509_bundle.py

```diff
@@ -168,90 +176,18 @@ def save(
-
-    result = set()
-    for cert in parsed_certs:
-        try:
-            x509_cert = x509.load_pem_x509_certificate(
-                cert.as_bytes(), default_backend()
-            )
-            result.add(x509_cert)
-        except Exception:
-            raise ParseX509BundleError('Unable to load PEM X.509 certificate')
-
-    return result
-
-
-def _parse_der_authorities(der_bytes: bytes) -> Set[Certificate]:
-    chain = set()
-    try:
-        leaf = x509.load_der_x509_certificate(der_bytes, default_backend())
-        chain.add(leaf)
-        _, remaining_data = decode(der_bytes)
-        while len(remaining_data) > 0:
-            cert = x509.load_der_x509_certificate(remaining_data, default_backend())
-            chain.add(cert)
-            _, remaining_data = decode(remaining_data)
-    except Exception as err:
-        raise ParseX509BundleError(str(err))
-
-    return chain
+        _write_bundle_to_file(bundle_path, encoding, x509_bundle)
 
 
-def _load_bundle_bytes(certs_chain_path: str) -> bytes:
-    try:
-        with open(certs_chain_path, 'rb') as chain_file:
-            return chain_file.read()
-    except FileNotFoundError:
-        raise LoadX509BundleError(
-            'Certs chain file file not found: {}'.format(certs_chain_path)
-        )
-    except Exception as err:
-        raise LoadX509BundleError(
-            'Certs chain file could not be read: {}'.format(str(err))
-        )
-
-
-def _write_certs_to_file(
+def _write_bundle_to_file(
     bundle_path: str,
     encoding: serialization.Encoding,
     x509_bundle: 'X509Bundle',
 ) -> None:
     try:
-        with open(bundle_path, 'wb') as chain_file:
-            os.chmod(chain_file.name, _BUNDLE_FILE_MODE)
+        with open(bundle_path, 'wb') as certs_file:
+            os.chmod(certs_file.name, _BUNDLE_FILE_MODE)
             for cert in x509_bundle.x509_authorities():
-                _write_cert_to_file(cert, chain_file, encoding)
-    except Exception as err:
-        raise SaveX509BundleError('Error opening certs chain file: {}'.format(str(err)))
-
-
-def _write_cert_to_file(
-    authority: Certificate,
-    bundle_file: BinaryIO,
-    encoding: serialization.Encoding,
-) -> None:
-    try:
-        authority_bytes = _extract_chain_bytes(authority, encoding)
-        bundle_file.write(authority_bytes)
+                write_certificate_to_file(cert, certs_file, encoding)
     except Exception as err:
-        raise SaveX509BundleError(
-            'Error writing authority certificate to file: {}'.format(str(err))
-        )
-
-
-def _extract_chain_bytes(cert: Certificate, encoding: serialization.Encoding) -> bytes:
-    try:
-        cert_bytes = cert.public_bytes(encoding)
-    except Exception as err:
-        raise X509BundleError('Could not get bytes from object: {}'.format(str(err)))
-
-    return cert_bytes
+        raise SaveX509BundleError('Error opening bundle file: {}'.format(str(err)))
```

Comment:
Error message here should be: `'Error writing certificate to file: {}'` (or similar), right?